dynamic.js

```diff
@@ -3,16 +3,30 @@
 const fs = require('fs')
 const path = require('path')
 const yaml = require('js-yaml')
+const Ref = require('json-schema-resolver')
 
 module.exports = function (fastify, opts, next) {
   fastify.decorate('swagger', swagger)
 
   const routes = []
+  const sharedSchemasMap = new Map()
+  const ref = Ref({ clone: true })
 
   fastify.addHook('onRoute', (routeOptions) => {
     routes.push(routeOptions)
   })
 
+  fastify.addHook('onRegister', async (instance) => {
+    await instance.ready() // wait the addSchema
```

Comment:
I'm a bit worried by this change. How would you expect this to work?

Reply:
onRegister is "fire and forget" and the await ready is blocked till the start

I should manage error

Reply:
Don't pass an `async` function here. Otherwise it would be completely unclear to the reader.

Reply:
isn't https://github.com/fastify/fastify/pull/2166 / `onReady` what would be best used?

Reply:
I thought it wasn't released yet, I will update 👍